packages/oft-evm/contracts/NativeOFTAdapter.sol

```diff
@@ -0,0 +1,130 @@
+     *  - nonce: The nonce of the sent message.
+     *  - fee: The LayerZero fee incurred for the message.
+     */
+    function send(
+        SendParam calldata _sendParam,
+        MessagingFee calldata _fee,
+        address _refundAddress
+    ) public payable virtual override returns (MessagingReceipt memory msgReceipt, OFTReceipt memory oftReceipt) {
+        // @dev Ensure the native funds in msg.value are exactly enough to cover the fees and amount to send (with dust removed).
+        // @dev This will revert if the _sendParam.amountLD contains any dust
+        uint256 requiredMsgValue = _fee.nativeFee + _removeDust(_sendParam.amountLD);
+        if (msg.value != requiredMsgValue) {
+            revert IncorrectMessageValue(msg.value, requiredMsgValue);
+        }
+
+        return _send(_sendParam, _fee, _refundAddress);
+    }
+
+    /**
+     * @dev Locks native sent by the sender as msg.value
+     * @dev _from The address to debit.
+     * @param _amountLD The amount of native to send in local decimals.
+     * @param _minAmountLD The minimum amount to send in local decimals.
+     * @param _dstEid The destination chain ID.
+     * @return amountSentLD The amount sent in local decimals.
+     * @return amountReceivedLD The amount received in local decimals on the remote.
+     */
+    function _debit(
+        address /*_from*/,
+        uint256 _amountLD,
+        uint256 _minAmountLD,
+        uint32 _dstEid
+    ) internal virtual override returns (uint256 amountSentLD, uint256 amountReceivedLD) {
+        // @dev Native funds sent with msg.value are locked into this contract higher up on the overridden send() function
+        (amountSentLD, amountReceivedLD) = _debitView(_amountLD, _minAmountLD, _dstEid);
+    }
+
+    /**
+     * @dev Credits native to the specified address.
+     * @param _to The address to credit the native to.
+     * @param _amountLD The amount of native to credit.
+     * @dev _srcEid The source chain ID.
+     * @return amountReceivedLD The amount of native ACTUALLY received.
+     */
+    function _credit(
+        address _to,
+        uint256 _amountLD,
+        uint32 /*_srcEid*/
+    ) internal virtual override returns (uint256 amountReceivedLD) {
+        // @dev Transfer tokens to the recipient.
+        (bool success, bytes memory data) = payable(_to).call{value: _amountLD}("");
+        if (!success) {
+            revert CreditFailed(_to, _amountLD, data);
+        }
+
+        // @dev In the case of NON-default NativeOFTAdapter, the amountLD MIGHT not be == amountReceivedLD.
+        return _amountLD;
+    }
+
+    /**
+     * @dev Overridden to be empty as this assertion is done higher up on the overriden send() function.
+     * @param _nativeFee The native fee to be paid.
+     * @return nativeFee The amount of native currency paid.
+     */
+    function _payNative(uint256 _nativeFee) internal pure override returns (uint256 nativeFee) {
```

Comment:
Why name the return parameter if you are not assigning it?

Reply:
That's a great point that I didn't think of! I will keep in mind that return variables should only be named if being assigned. Considering this has already been audited, I am hesitant to change any code even though I removed the return parameter name locally and the tests still passed

Please let me know if you strongly recommend this change, if so I will look into whether the auditors are okay with this minor change 🫡

Reply:
No it is fine, just wanted to make sure I wasn't overlooking anything.